src/main/java/at/hannibal2/skyhanni/features/event/hoppity/HoppityEventSummary.kt

```diff
@@ -62,6 +174,128 @@ object HoppityEventSummary {
+                "§6§lReminder! §r§eSwitch to a new server and run §6/cf §e to " +
+                    "update your leaderboard position in Hoppity Event stats."
+            )
+        }
+    }
+
+    private fun buildDisplayRenderables(stats: HoppityEventStats?, statYear: Int): List<Renderable> = buildList {
+        // Add title renderable with centered alignment
+        add(
+            Renderable.string(
+                "§dHoppity's Hunt #${getHoppityEventNumber(statYear)} Stats",
+                horizontalAlign = RenderUtils.HorizontalAlignment.CENTER
+            )
+        )
+
+        // Conditionally add year switcher renderable for inventory or chest screens
+        if (isInInventory()) {
+            buildYearSwitcherRenderables(statYear)?.let { yearSwitcher ->
+                add(
+                    Renderable.horizontalContainer(
+                        yearSwitcher,
+                        spacing = 5,
+                        horizontalAlign = RenderUtils.HorizontalAlignment.CENTER
+                    )
+                )
+            }
+        }
+
+        // Add card renderable based on stats availability
+        val cardRenderable = if (stats == null) {
+            Renderable.verticalContainer(
+                mutableListOf(Renderable.string("§cNo stats found for Hunt #${getHoppityEventNumber(statYear)}."))
+            )
+        } else {
+            Renderable.verticalContainer(
+                getStatsStrings(stats, statYear).map { Renderable.string(it.string) }.toMutableList()
+            )
+        }
+        add(cardRenderable)
+    }
+
+    private fun buildYearSwitcherRenderables(currentStatYear: Int): List<Renderable>? {
+        val profileStorage = ProfileStorageData.profileSpecific ?: return null
+        val statsStorage = profileStorage.hoppityEventStats
+        val statsYearList = statsStorage.keys.takeIf { it.isNotEmpty() } ?: mutableListOf()
+
+        val predecessorYear = statsYearList.filter { it < currentStatYear }.takeIf { it.any() }?.max()
+        val successorYear = statsYearList.filter { it > currentStatYear }.takeIf { it.any() }?.min()
+        if (predecessorYear == null && successorYear == null) return null
+
+        return listOfNotNull(
+            predecessorYear?.let {
+                Renderable.optionalLink(
+                    "§d[ §r§f§l<- §r§7Hunt #${getHoppityEventNumber(it)} §r§d]",
+                    onClick = { profileStorage.hoppityStatLiveDisplayYear = it }
+                )
+            },
+            successorYear?.let {
+                Renderable.optionalLink(
+                    "§d[ §7Hunt #${getHoppityEventNumber(it)} §r§f§l-> §r§d]",
+                    onClick = { profileStorage.hoppityStatLiveDisplayYear = it }
+                )
+            }
+        )
+    }
+
     private fun checkInit() {
         val statStorage = ProfileStorageData.profileSpecific?.hoppityEventStats ?: return
         val currentYear = SkyBlockTime.now().year
```

Comment:
isntead of handling creation of a new hoppity event stats instance here, make a new val with a getter for currentYearStorage or similar

something like this:

```kt
    private val currentYearStorage: ProfileSpecificStorage.HoppityEventStats?
        get() = storage?.getOrPut(SkyBlockTime.now().year, ::HoppityEventStats)
```